src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -305,7 +305,7 @@ protected override Encoding StandardOutputEncoding
 
         protected override int ExecuteTool(string pathToTool, string responseFileCommands, string commandLineCommands)
         {
-            if (!UseSharedCompilation || this.ToolPath != null )
+            if (!UseSharedCompilation || !String.IsNullOrEmpty(this.ToolPath))
```

Comment:
I disagree -- empty CscToolPath is empty CscToolPath, not the magic currenty directory. MSBuild's system of defining properties has very little difference between undefined properties and empty properties. While this may have been a poor design decision, it's already endemic in the system.